### PR TITLE
Fix conflicting npm dependencies and bump typescript 4 to 5

### PR DESCRIPTION
**Pull request by shivam-sharma7**
## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: horreum-web/package.json [hunk 1]
```diff
@@ -21,6 +22,7 @@
     "clsx": "1.2.1",
     "fetchival": "0.3.3",
     "github-markdown-css": "5.5.0",
+    "history": "4.10.1",
```
Comment (johnaohara) [inline horreum-web/package.json:25]: I removed the dependency on `history` when I upgrade from react-router v5 to v6 in this PR https://github.com/Hyperfoil/Horreum/pull/1214/files

"react-router": "6.21.3" has a transitive dependency on `history`, and pulls in the correct version for it to function 

Please remove this dependency

File: horreum-web/package.json [hunk 2]
```diff
@@ -8,6 +8,7 @@
     "@patternfly/patternfly": "5.1.0",
     "@patternfly/react-core": "5.1.1",
     "@patternfly/react-table": "5.1.1",
+    "@types/history": "4.7.11",
```
Comment (johnaohara) [inline horreum-web/package.json:11]: please see comment below about "history": "4.10.1",

Please remove this dependency

**Review by johnaohara — CHANGES_REQUESTED** [review]

File: horreum-web/package.json (hunk 1 above)
Comment (shivam-sharma7) [reply horreum-web/package.json:25]: @johnaohara see-https://github.com/Hyperfoil/Horreum/blob/5e274ae3d779e652cb7a29a0e0e1711d3368ae82/horreum-web/src/store.ts#L1

I'm getting error during compile, this is why I have to do this which is mostly right

File: horreum-web/package.json (hunk 2 above)
Comment (shivam-sharma7) [reply horreum-web/package.json:11]: read above comment

File: horreum-web/package.json (hunk 1 above)
Comment (johnaohara) [reply horreum-web/package.json:25]: > I'm getting error during compile
Please can you share details/log of the error?
You should not need an explicit dependency on "history" as it is a transitive dependency from "react-router"  (in addition "history": "4.10.1" is older than the history dependency that "react-router" pulls in)

Comment (shivam-sharma7) [reply horreum-web/package.json:25]: @johnaohara look at the following, its hapening cause of https://github.com/Hyperfoil/Horreum/pull/1214/files#diff-e088454949d6ee9a87cc5bd8e7698ee60b57471d69780fb6bd132d95981ed7ef
```java
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerRunner] Run `npm audit` for details.
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerRunner] Running Quinoa package manager build command: node.exe "C:\Users\shivam sharma\Desktop\My Project\open-source\Horreum\horreum-web\node\node\node_modules\npm\bin\npm-cli.js" run build
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerRunner] 
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerRunner] > horreum-web@0.1.0 build
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerRunner] > tsc && vite build
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerRunner]
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManagerRunner] src/store.ts(1,38): error TS2307: Cannot find module 'history' or its corresponding type declarations.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Horreum 0.12-SNAPSHOT:
[INFO]
[INFO] Horreum ............................................ SUCCESS [  0.850 s]
[INFO] horreum-api ........................................ SUCCESS [ 11.138 s]
[INFO] Horreum Client ..................................... SUCCESS [  1.061 s]
[INFO] Horreum infra - common ............................. SUCCESS [  0.341 s]
[INFO] Horreum Dev Services - Parent ...................... SUCCESS [  0.035 s]
[INFO] Horreum Dev Services - Runtime ..................... SUCCESS [  2.310 s]
[INFO] Horreum Dev Services - Deployment .................. SUCCESS [  3.308 s]
[INFO] Horreum Backend .................................... FAILURE [ 48.555 s]
[INFO] Horreum Integration Tests .......................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:09 min
[INFO] Finished at: 2024-02-01T08:40:18+05:30
[INFO] ------------------------------------------------------------------------

Comment (johnaohara) [reply horreum-web/package.json:25]: This is what I am struggling to understand.  

For that PR ([1]), the build passes:  [2]

And in the package-lock.json for that PR, there still is the history module: [3] . Also, notice the versions for history "version": "4.10.1" ->  "version": "5.3.0"

History should still be in node_modules, even though the explicit dependency has been removed

What happens when you run;
`mvn clean -Premove-node-cache` 
then 
`mvn clean install -DskipTests -DskipITs`?


Can you share the output of `npm ls --all` from you branch where the build is failing?

You can see in the output in my env ([4]) that the history module is still in the dependency tree [5] , and therefore the module should still be present at compile time.


[1]  - https://github.com/Hyperfoil/Horreum/pull/1214/checks
[2] - https://github.com/Hyperfoil/Horreum/pull/1214/checks#step:6:602
[3] - https://github.com/Hyperfoil/Horreum/pull/1214/files#diff-7bd5bcb1bbc7060dbc4c33ca2164d862273f7518582df29baefca7f356f8daaaR2893 
[4] - https://gist.github.com/johnaohara/b0bbefe71855712b7133d94f91e3ba56
[5] - https://gist.github.com/johnaohara/b0bbefe71855712b7133d94f91e3ba56#file-gistfile1-txt-L657

Comment (johnaohara) [reply horreum-web/package.json:25]: > @johnaohara see-
> 
> https://github.com/Hyperfoil/Horreum/blob/5e274ae3d779e652cb7a29a0e0e1711d3368ae82/horreum-web/src/store.ts#L1
> 
> I'm getting error during compile, this is why I have to do this which is mostly right

I think we can remove all of this

**Comment by johnaohara** [comment]
> @johnaohara let's merge it

@shivam-sharma7 , I am really sorry, I merged in another PR before this and there is a conflict with package-lock.json again

I am going to fix the package-lock.json issue and mere.

We seem to have a problem with some PR's that the entire package-lock.json is re-written with a different number of spaces, which means the entire file is replaced

**Review by johnaohara — APPROVED** [review]

**Comment by johnaohara** [comment]
Merged, Thanks for your contribution.

**Pull request merged**